node/test/trace/basic_server.js

```diff
@@ -70,17 +70,27 @@ test('basic tracing test', function (assert) {
     server.handler.register('/top_level_endpoint', function (req, res) {
         logger.info("top level sending to subservice");
         setTimeout(function () {
-            server
-                .request({
-                    host: '127.0.0.1:4042', 
-                    serviceName: 'subservice', 
-                    parentSpan: req.span,
-                    trace: true
-                }).send('/foobar', 'arg1', 'arg2', function (err, subRes) {
-                    logger.info("top level recv from subservice");
+            var servReq = server.request({
+                host: '127.0.0.1:4042',
+                serviceName: 'subservice',
+                parentSpan: req.span,
+                trace: true});
+            var peers = server.peers.values();
+            var ready = new CountedReadySignal(peers.length);
+            peers.forEach(function each(peer) {
```

Comment:
This would be another candidate for a peer.withConnection(ready.signal) utility to DRY it up.